In [6]:
#install packages with pip
#pip install pandas numpy

In [7]:
# Import packages for data manipulation
import pandas as pd
import numpy as np

In [10]:
# Load csv dataset into dataframe // https://www.kaggle.com/datasets/mehmetsabrikunt/internet-service-churn
df = pd.read_csv('internet_service_churn.csv')

In [18]:
df.head()


id  is_tv_subscriber  is_movie_package_subscriber  subscription_age  \
0  15                 1                            0             11.95   
1  18                 0                            0              8.22   
2  23                 1                            0              8.91   
3  27                 0                            0              6.87   
4  34                 0                            0              6.39   

   bill_avg  reamining_contract  service_failure_count  download_avg  \
0        25                0.14                      0           8.4   
1         0                 NaN                      0           0.0   
2        16                0.00                      0          13.7   
3        21                 NaN                      1           0.0   
4         0                 NaN                      0           0.0   

   upload_avg  download_over_limit  churn  
0         2.3                    0      0  
1         0.0                    0      1  
2         0.9                    0      1  
3         0.0                    0      1  
4         0.0                    0      1

In [19]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 72274 entries, 0 to 72273
Data columns (total 11 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   id                           72274 non-null  int64  
 1   is_tv_subscriber             72274 non-null  int64  
 2   is_movie_package_subscriber  72274 non-null  int64  
 3   subscription_age             72274 non-null  float64
 4   bill_avg                     72274 non-null  int64  
 5   reamining_contract           50702 non-null  float64
 6   service_failure_count        72274 non-null  int64  
 7   download_avg                 71893 non-null  float64
 8   upload_avg                   71893 non-null  float64
 9   download_over_limit          72274 non-null  int64  
 10  churn                        72274 non-null  int64  
dtypes: float64(4), int64(7)
memory usage: 6.1 MB


In [41]:
# Isolate rows where churn is 1
churn_df = df[df['churn'] == 1]
# Display summary stats of rows with churn values
churn_df.describe()

id  is_tv_subscriber  is_movie_package_subscriber  \
count  4.005000e+04      40050.000000                 40050.000000   
mean   6.483943e+05          0.701223                     0.204619   
std    4.322926e+05          0.457727                     0.403428   
min    1.800000e+01          0.000000                     0.000000   
25%    2.863745e+05          0.000000                     0.000000   
50%    5.907300e+05          1.000000                     0.000000   
75%    9.498655e+05          1.000000                     0.000000   
max    1.689744e+06          1.000000                     1.000000   

       subscription_age      bill_avg  reamining_contract  \
count      40050.000000  40050.000000        20331.000000   
mean           2.226554     18.692859            0.183152   
std            1.709438     13.073794            0.458280   
min           -0.020000      0.000000            0.000000   
25%            0.980000     13.000000            0.000000   
50%            1.970000     20.000000            0.000000   
75%            2.940000     23.000000            0.000000   
max           12.800000    278.000000            2.310000   

       service_failure_count  download_avg    upload_avg  download_over_limit  \
count           40050.000000  40050.000000  40050.000000         40050.000000   
mean                0.290037     26.838312      2.767698             0.349213   
std                 0.865134     46.137197      7.932245             1.290545   
min                 0.000000      0.000000      0.000000             0.000000   
25%                 0.000000      0.000000      0.000000             0.000000   
50%                 0.000000     11.900000      0.900000             0.000000   
75%                 0.000000     36.700000      2.900000             0.000000   
max                19.000000   1706.200000    327.200000             7.000000   

         churn  
count  40050.0  
mean       1.0  
std        0.0  
min        1.0  
25%        1.0  
50%        1.0  
75%        1.0  
max        1.0

In [42]:
# Isolate rows where churn is 0
no_churn_df = df[df['churn'] == 0]
# Display summary stats of rows with churn values
no_churn_df.describe()


id  is_tv_subscriber  is_movie_package_subscriber  \
count  3.222400e+04      32224.000000                 32224.000000   
mean   1.092310e+06          0.956989                     0.496214   
std    4.416807e+05          0.202886                     0.499993   
min    1.500000e+01          0.000000                     0.000000   
25%    8.168338e+05          1.000000                     0.000000   
50%    1.189630e+06          1.000000                     0.000000   
75%    1.459718e+06          1.000000                     1.000000   
max    1.680282e+06          1.000000                     1.000000   

       subscription_age      bill_avg  reamining_contract  \
count      32224.000000  32224.000000        30371.000000   
mean           2.727828     19.252731            1.072765   
std            2.348857     13.382995            0.594408   
min            0.000000      0.000000            0.000000   
25%            0.880000     13.000000            0.560000   
50%            2.050000     18.000000            1.080000   
75%            4.030000     22.000000            1.620000   
max           11.950000    406.000000            2.920000   

       service_failure_count  download_avg    upload_avg  download_over_limit  \
count           32224.000000  31843.000000  31843.000000         32224.000000   
mean                0.254593     64.884728      5.983563             0.031622   
std                 0.751527     74.788156     11.522778             0.322663   
min                 0.000000      0.000000      0.000000             0.000000   
25%                 0.000000     26.000000      1.900000             0.000000   
50%                 0.000000     48.800000      3.700000             0.000000   
75%                 0.000000     84.100000      6.700000             0.000000   
max                16.000000   4415.200000    453.300000             7.000000   

         churn  
count  32224.0  
mean       0.0  
std        0.0  
min        0.0  
25%        0.0  
50%        0.0  
75%        0.0  
max        0.0

In [40]:
# Isolate rows with null values
null_df = df[df['reamining_contract'].isnull()]
# Display summary stats of rows with null values
null_df.describe()

id  is_tv_subscriber  is_movie_package_subscriber  \
count  2.157200e+04      21572.000000                 21572.000000   
mean   5.178292e+05          0.630910                     0.183710   
std    3.892458e+05          0.482569                     0.387257   
min    1.800000e+01          0.000000                     0.000000   
25%    1.841980e+05          0.000000                     0.000000   
50%    4.485905e+05          1.000000                     0.000000   
75%    7.630318e+05          1.000000                     0.000000   
max    1.689600e+06          1.000000                     1.000000   

       subscription_age      bill_avg  reamining_contract  \
count      21572.000000  21572.000000                 0.0   
mean           2.807239     20.484888                 NaN   
std            1.805538     18.486054                 NaN   
min            0.000000      0.000000                 NaN   
25%            1.910000     13.000000                 NaN   
50%            2.310000     21.000000                 NaN   
75%            3.550000     24.000000                 NaN   
max           12.800000    406.000000                 NaN   

       service_failure_count  download_avg    upload_avg  download_over_limit  \
count           21572.000000  21518.000000  21518.000000         21572.000000   
mean                0.275450     26.134511      2.713310             0.520119   
std                 0.843656     68.532786      8.776637             1.564586   
min                 0.000000      0.000000      0.000000             0.000000   
25%                 0.000000      0.000000      0.000000             0.000000   
50%                 0.000000      9.100000      0.700000             0.000000   
75%                 0.000000     30.300000      2.500000             0.000000   
max                19.000000   3890.800000    280.700000             7.000000   

              churn  
count  21572.000000  
mean       0.914102  
std        0.280220  
min        0.000000  
25%        1.000000  
50%        1.000000  
75%        1.000000  
max        1.000000

In [26]:
# Isolate rows without null values
not_null_df = df[~df['reamining_contract'].isnull()]
# Display summary stats of rows without null values
not_null_df.describe()

id  is_tv_subscriber  is_movie_package_subscriber  \
count  5.070200e+04      50702.000000                 50702.000000   
mean   9.860793e+05          0.893693                     0.398840   
std    4.594533e+05          0.308234                     0.489665   
min    1.500000e+01          0.000000                     0.000000   
25%    6.256515e+05          1.000000                     0.000000   
50%    1.033207e+06          1.000000                     0.000000   
75%    1.384116e+06          1.000000                     1.000000   
max    1.689744e+06          1.000000                     1.000000   

       subscription_age      bill_avg  reamining_contract  \
count      50702.000000  50702.000000        50702.000000   
mean           2.298080     18.286241            0.716039   
std            2.106847     10.105378            0.697102   
min           -0.020000      0.000000            0.000000   
25%            0.760000     13.000000            0.000000   
50%            1.600000     18.000000            0.570000   
75%            3.160000     22.000000            1.310000   
max           12.760000    386.000000            2.920000   

       service_failure_count  download_avg    upload_avg  download_over_limit  \
count           50702.000000  50375.000000  50375.000000         50702.000000   
mean                0.273717     51.188812      4.823740             0.074652   
std                 0.804851     59.527979     10.166523             0.562646   
min                 0.000000      0.000000      0.000000             0.000000   
25%                 0.000000     14.600000      1.000000             0.000000   
50%                 0.000000     37.200000      2.800000             0.000000   
75%                 0.000000     70.400000      5.600000             0.000000   
max                18.000000   4415.200000    453.300000             7.000000   

              churn  
count  50702.000000  
mean       0.400990  
std        0.490104  
min        0.000000  
25%        0.000000  
50%        0.000000  
75%        1.000000  
max        1.000000

In [38]:
# Calculate counts of churned vs. retained
print(df['reamining_contract'].notnull().value_counts())
print()
print(df['reamining_contract'].notnull().value_counts(normalize=True))

reamining_contract
True     50702
False    21572
Name: count, dtype: int64

reamining_contract
True     0.701525
False    0.298475
Name: proportion, dtype: float64


## Null Values of Remaining contract
For Non-Null values of remaining contract, churn is higher. If a customer has a remaining contract, the churn is higher. 50702 have non-null, and 21572 have null. 

That means 0.298475 have null and 0.701525 have not null

In [36]:
# Calculate counts of churned vs. retained
print(df['churn'].value_counts())
print()
print(df['churn'].value_counts(normalize=True))

churn
1    40050
0    32224
Name: count, dtype: int64

churn
1    0.554141
0    0.445859
Name: proportion, dtype: float64
